### DISCUSSION TITLE
How to test a model on testing dataset ?

### DISCUSSION THREAD

**Discussion by HripsimeS** [Project-MONAI/MONAI#3875] [Q&A] [answered]
Hello. I am trying to use this notebook tutorial for my dataset and for the training I received quite good results best score of 96%.
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d.ipynb

I noticed the size of your dataset was 61 3D volumes (41 Training + 20 Testing). So you used 32 cases for training, 9 cases for validation and normally 20 cases should be for testing. In the end of notebook there is an "evaluation of model on original image spacings" where as I understood you used validation 9 cases.
Can you please give me more information how to test the model on those 20 testing cases located in "imagesTs" folder, because I want to test the model on dataset that I did not use during training. Thanks in advance and look forward to hearing from you!

**Comment by yiheng-wang-nv** [comment, marked as answer]
Hi @HripsimeS , do inference is very close to do validation, except the calculating metrics/losses part.
You may need to set transforms that do not include `"label"`.
You can also refer to this tutorial:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_inference_dict.py

**Reply by HripsimeS** [reply]
Hi @yiheng-wang-nv. As you have testing 20 images in Spleen dataset, it would be nice to include in tutorial notebook an evaluation of model of test images to make sure that the model is working properly. I believe in segmentation, evaluation of model on testing data is even more important than the evaluation of model on original image spacings.

**Reply by yiheng-wang-nv** [reply]
Hi @HripsimeS , let me add the inference part later in tutorial. Thanks for the advice : )

**Reply by HripsimeS** [reply]
Hello @yiheng-wang-nv. Do you have some news for inference part, is it already added in the tutorial? I hope it would be possible to have an inference part in tutorial during this week 🥇

**Reply by yiheng-wang-nv** [reply]
Hi @HripsimeS , please track [here](https://github.com/Project-MONAI/tutorials/pull/595)

**Reply by HripsimeS** [reply]
@yiheng-wang-nv thank you so much!! 👍

**Comment by rijobro** [comment]
@HripsimeS converted to a discussion on the MONAI main repository.
